internal/local/structure_v2.go

```diff
@@ -331,3 +327,204 @@ func parseSync(rootDir string) (*SyncStructure, error) {
 	}
 	return &SyncStructure{config}, nil
 }
+
+// ConfigData marshals the config data out to JSON
+func (a AppDataV2) ConfigData() ([]byte, error) {
+	temp := &struct {
+		ConfigVersion         realm.AppConfigVersion `json:"config_version"`
+		ID                    string                 `json:"app_id,omitempty"`
+		Name                  string                 `json:"name,omitempty"`
+		Location              realm.Location         `json:"location,omitempty"`
+		DeploymentModel       realm.DeploymentModel  `json:"deployment_model,omitempty"`
+		Environment           string                 `json:"environment,omitempty"`
+		AllowedRequestOrigins []string               `json:"allowed_request_origins,omitempty"`
+	}{
+		ConfigVersion:         a.ConfigVersion(),
+		ID:                    a.ID(),
+		Name:                  a.Name(),
+		Location:              a.Location(),
+		DeploymentModel:       a.DeploymentModel(),
+		Environment:           a.Environment,
+		AllowedRequestOrigins: a.AllowedRequestOrigins,
+	}
+	return MarshalJSON(temp)
+}
+
+// WriteData will write the local Realm app data to disk
+func (a AppDataV2) WriteData(rootDir string) error {
+	if err := writeSecrets(rootDir, a.Secrets); err != nil {
+		return err
+	}
+	if err := writeEnvironments(rootDir, a.Environments); err != nil {
+		return err
+	}
+	if err := writeValues(rootDir, a.Values); err != nil {
+		return err
+	}
+	graphQL := a.GraphQL
+	if a.GraphQL == nil {
+		graphQL = &GraphQLStructure{}
+	}
+	if err := writeGraphQL(rootDir, *graphQL); err != nil {
+		return err
+	}
+	if err := writeServices(rootDir, a.Services); err != nil {
+		return err
+	}
+	if err := writeFunctionsV2(rootDir, a.Functions); err != nil {
+		return err
+	}
+	if err := writeAuth(rootDir, a.Auth); err != nil {
+		return err
+	}
+	if err := writeSync(rootDir, a.Sync); err != nil {
+		return err
+	}
+	if err := writeDataSources(rootDir, a.DataSources); err != nil {
+		return err
+	}
+	if err := writeHTTPEndpoints(rootDir, a.HTTPEndpoints); err != nil {
+		return err
+	}
+	return nil
+}
+
+func writeFunctionsV2(rootDir string, functions *FunctionsStructure) error {
+	configs := []map[string]interface{}{}
+	if functions != nil {
+		configs = functions.Configs
+	}
+	dataFuncs, err := MarshalJSON(configs)
+	if err != nil {
+		return err
+	}
+	err = WriteFile(filepath.Join(rootDir, NameFunctions, FileConfig.String()), 0666, bytes.NewReader(dataFuncs))
+	if err != nil {
+		return err
+	}
+	if functions == nil {
+		return nil
```

Comment:
id either add a comment here why we're returning early (`// if no functions, write the empty config and no more`), or use a similar pattern above:

```go
var sources map[string]string
if functions != nil {
  sources = functions.Sources // you could even do this all at the top of the function in one "if not nil" check
}
```